polar-core/src/error.rs

```diff
@@ -82,47 +160,33 @@ impl From<PolarError> for FormattedPolarError {
     fn from(other: PolarError) -> Self {
         Self {
             formatted: other.to_string(),
-            kind: other.kind,
+            kind: other.0,
         }
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub enum ErrorKind {
-    Parse(ParseError),
-    Runtime(RuntimeError),
-    Operational(OperationalError),
-    Validation(ValidationError),
+#[derive(Clone, Debug, Serialize)]
+#[serde(transparent)]
+pub struct ParseError {
+    pub kind: ParseErrorKind,
+    #[serde(skip_serializing)]
+    pub source: Arc<Source>,
 }
```

Comment:
Made this change in 30409c9a89e1a78b8de8ac34c0f0c6ab85e516d6 to avoid duplicating the `source: Arc<Source>` field on every single `ParseError` variant, but I'm not sure if avoiding that is worth the added complexity 🤷